vignettes/presize.Rmd

```diff
@@ -0,0 +1,189 @@
+
+```{r}
+prec_mean(22, sd = 10, conf.width = 5)
+```
+
+### Multiple scenarios
+
+It is common for only a vague idea of what to expect in terms of SDs, sensitivities, etc, so it is often worthwhile creating a set of scenarios.
+
+Returning to the colposcope example from above... We want to see how the CI width varies with different sensitivities. With `presize`, it's easy to run different scenarios, simply by passing multiple values to each parameter (where multiple values are passed, they should have the same length!). For instance, for varying a single parameter, scenarios we want to look at can be created with `seq` and passed that to the appropriate `presize` functions argument. Here we vary sensitivity between 50% and 95% in steps of 5%.
+
+```{r}
+(scenario_data <- prec_sens(sens = seq(.5, .95, .05), 
+                            prev = .15, 
+                            ntot = 250, 
+                            method = "wilson"))
+
+```
+
+We can also use `expand.grid` to pass scenarios varying multiple parameters simultaneously. Below we vary sensitivity, prevalence and sample size.
+
+```{r}
+scenarios <- expand.grid(sens = seq(.5, .95, .05),
+                         prev = seq(.1, .2, .02),
+                         ntot = c(250, 350))
+
+(scenario_data <- prec_sens(sens = scenarios$sens, 
+                            prev = scenarios$prev, 
+                            ntot = scenarios$ntot, 
+                            method = "wilson"))
+```
+
+From the print method, we see the details of the individual scenarios. The print method for `presize` objects only prints the first 10 rows, but there are in fact `r length(scenario_data$sens)` rows. 
+
+Using the `as.data.frame` method, we can convert the list returned by `prec_sens` to a dataframe from which we can create tables or figures. Where multiple scenarios are calculated, plotting them can be particularly informative. `ggplot2`, for example, is particularly useful for this. Below we show CI width as a function of sensitivity, but the other parameters could be chosen instead.
+
+```{r, fig.width=7}
+scenario_df <- as.data.frame(scenario_data)
+
+library(ggplot2)
+
+ggplot(scenario_df, 
+       aes(x = sens, 
+           y = conf.width, 
+           # convert colour to factor for distinct colours rather than a continuum
+           col = as.factor(prev), 
+           group = prev)) +
+  geom_line() +
+  labs(x = "Sensitivity", y = "CI width", col = "Prevalence") + 
+  facet_wrap(vars(ntot))
+
+```
+
+We could also create a table of the scenarios containing the CIs. Below we select only the scenarios with sensitivities above 70% and reshape and format the table and use the `gt` package to print a nice table in HTML. 
+
+```{r}
+library(dplyr)
+library(tidyr)
+library(magrittr)
+library(gt)
+
+scenario_df %>% 
+  # create the values needed specifically for the table
+  mutate(
+    txt = sprintf("%3.1f - %3.1f", lwr * 100, upr * 100),
+    Prevalence = prev * 100,
```

Comment:
Isn't it possible to display the header Prevalence more clearly regarding the prevalence numbers? E.g. vertically in the left margin...